### PR TITLE
feat: `renderFooterInTable` toggle to render footer as `<tfoot />`

### PR DESCRIPTION
**Pull request by hackape**

### CODE REVIEW

File: examples/renderFooterInTable.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+            case 'height':
+              text = currentData.reduce((acc, record) => acc + record[dataIndex], 0) / currentData.length;
+              text = 'Average Height: ' + text.toFixed(2);
+              break;
+          }
+
+          return <Cell key={key || dataIndex}>{text}</Cell>
+        })
+
+        return <Row>{cells}</Row>
```
Comment (yesmeck) [inline examples/renderFooterInTable.js]: Is returning `<Row>` required?

Comment (hackape) [reply examples/renderFooterInTable.js]: Not really. Only when you set `renderFooterInTable` to true should you return `<Row />`.

Comment (hackape) [reply examples/renderFooterInTable.js]: I keep the original `footer` API untouched. Pass extra args to it, but only takes effect when you toggle `renderFooterInTable` on.

**Comment by hackape** [comment]
I only intent to provide a way to put the footer right inside the table's `<tfoot />`. User can still return whatever they want from `footer={(...args) => Element}`. If `renderFooterInTable === true`, that `<Element />` will be insert into `<tfoot />` .

As of `args`, it contains `(currentData, columns, Row, Cell)`. `columns` includes an extra `{ key: 'expand-icon-placeholder' }` when needed, `Row` is `components.footer.row`, `Cell` the same. It's still up to user to construct the right return value. One can still put an `<div />` into `<tfoot />`, react will complain, but browser will let it through.

**Comment by hackape** [comment]
@yesmeck Any feedback?

File: src/FootTable.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+  return (
+    <div
+      key="footTable"
+      ref={fixed ? null : saveRef('footTable')}
+      className={`${prefixCls}-footer`}
+      style={footStyle}
+      onScroll={handleBodyScrollLeft}
+    >
+      <BaseTable
```
Comment (yesmeck) [inline src/FootTable.js]: Why render `tfoot` to a separator `BaseTable`, can we just render it in the `BaseTable`?

**Comment by yesmeck** [comment]
screen shot 2018-02-27 at 14 47 36

It seems to render `tfoot` to a sparator `BaseTable` break the hover state syncing.

File: src/FootTable.js (hunk 2 above)
Comment (hackape) [reply src/FootTable.js]: Reason is to keep its behavior consistent to Header's. When you turn `{ scroll: y }` on, Header is fixed, most likely user want the Footer to be fixed too.

**Comment by hackape** [comment]
Hmm, I didn't notice there's hover state syncing in there...🤔 Lemme check.

**Comment by yesmeck** [comment]
Could you consider the proposal I wrote in https://github.com/ant-design/ant-design/issues/5710#issuecomment-368768114

**Comment by hackape** [comment]
I see. Actually, I don't think there should be any hover state on footer at all. Current behavior is simply an side effect of css `tr:hover` style. Should just limit css to take effect on `tbody > tr` only, leave hover state on footer to user.

**Comment by hackape** [comment]
换中文吧？我同意你的这个 API 加入方式更好些。我会改成用这个来实现。

除了这点，现在的这个 fixed footer 的实现你有什么要 comment 的吗？

**Comment by hackape** [comment]
另外我建议命名为 `column = { tfoot: (data) => }` 更好吧，进一步避免跟 footer 的混淆。

**Comment by yesmeck** [comment]
就叫 “footer” 好了，我们也没有把 “row” 叫 “tr”。

**Comment by hackape** [comment]
ok，命名你们定。

**Comment by yesmeck** [comment]
其他看起来 ok

**Comment by yesmeck** [comment]
又考虑了下，`footer` 其实对应的就是 `title`。所以 `footer` 应该也支持直接写 `ReactNode`：

```javascript
const columns = [{
  title: 'Price',
  footer: `Amount: ${amount}`,
}]
```

**Comment by Nokecy** [comment]
@hackape  这个功能咋样呀

**Comment by hackape** [comment]
这段时间忙就给放下了，我今明两天应该能提个新的上来。

**Comment by xgj1988** [comment]
@hackape  column  footer  这个是否 可以在antd table 里面使用？

**Comment by hackape** [comment]
@xgj1988 这个 PR 被 revert 掉了，所以现在没有这功能。官方建议用自定义的 `table.props.components.body.wrapper` 来解决这问题。

**Comment by xgj1988** [comment]
@hackape  用哪个也有问题。 https://codesandbox.io/s/xjpkx9rzzw 这是别人的demo 。这里有两个问题。
1：合计列不能固定在底部。2：加上fixed 之后，样式全完乱了。

**Comment by hackape** [comment]
@xgj1988 不是让你直接用这个 demo 的实现，这只是给你提供一个可行思路。现在的情况就是官方团队不想增加 table 的复杂度，这种高级用法需要你自己去实现，而相应的接口就是 `table.props.components.body.wrapper`.

**Comment by bianlin** [comment]
打印需要用到tfoot，官方却没有这种打印需求。

**Comment by hackape** [comment]
@leejaen 你也可以尝试提一个完善的 PR

**Pull request closed without merging**